base/libgit2/callbacks.jl

```diff
@@ -82,7 +82,8 @@ function credentials_callback(cred::Ptr{Ptr{Void}}, url_ptr::Cstring,
         end
 
         # first try ssh-agent if credentials support its usage
-        if creds[:usesshagent, credid] == "Y"
+        if creds[:usesshagent, credid] === nothing ||
```

Comment:
Yeah, having a setindex that is later not retrievable by setindex with the same index violates expectations. It would be good to redesign that API